### ISSUE TITLE
`code tunnel` service should be a headless service

### ISSUE THREAD

**Issue by Myp3a** [microsoft/vscode#184058]
# Issue
`code tunnel` opens in a `cmd` window instead of running in background.

I have seen #167741, which introduced this behavior. For me, `service` means `run in background all the time without interacting with user`. Console window breaks it completely. If the window is closed - service is gone, what shouldn't happen.
I understand that there were reasons about changing service behavior to ordinary application. However, I would really like service-like functionality back for my remote development. It's perfectly fine if this would be an additional installation flag. That would keep current easy behavior and allow for unattended installs.

Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.78.2 (System)
- OS Version: Windows 11 22H2

Steps to Reproduce:

1. Run `code service install`
2. Reboot

**Comment by connor4312** [comment]
You can use `code tunnel service install` for that purpose 🙂

**Comment by kernel-sanders** [comment]
I believe this issue is only partially fixed, even in `code-cli 1.80.0 (commit 4649b5333370139d161e7b5d4f78ca13741efa35)`. The cmd.exe window does not appear when only `code.exe` or `code-insiders.exe` (the cli) is running, however, as soon as a tunnel is connected to, a cmd.exe window appears (as code-server.cmd is run). If the user closes the window, the tunnel connection is dropped.

I believe the fix is to pass `CREATE_NO_WINDOW` to creation_flags in `spawn_server_process` as seen here: https://github.com/kernel-sanders/vscode/commit/fd89ea0423bccce0df5a38f8956bfe0aac88c3f7. However, without the proprietary "mixin" product.json, it seems my builds of code cli are not able to properly run code-server, failing with

```
[2023-06-28 19:09:25] info [rpc.3] Starting server...
[2023-06-28 19:09:25] debug [rpc.3] Starting server with command... Command { std: "C:\\Users\\redacted\\.vscode-insiders\\cli\\servers\\Stable-695af097c7bd098fbf017ce3ac85e09bbc5dda06\\server\\bin\\code-server.cmd" "--connection-token=redacted" "--accept-server-license-terms" "--start-server" "--enable-remote-auto-shutdown" "--socket-path=\\\\.\\pipe\\code-oss-8cf2f48a-ab53-485a-bc7d-0bfba9f481fd", kill_on_drop: false }
[2023-06-28 19:09:25] trace [rpc.3] [codeserver.13] The syntax of the command is incorrect.
[2023-06-28 19:09:26] debug [tunnels::connections::ws] received liveness pong
```

To reproduce the issue, on a windows machine:

1. Create an arbitrary tunnel to authorize the machine: `code-insiders.exe tunnel --name test`
2. Authorize the tunnel with the code at https://github.com/login/device
3. Install the tunnel service with `code-insiders.exe service tunnel install`
4. Reboot
5. On https://vscode.dev connect to the tunnel again
6. Observe a cmd.exe on the desktop, close it and observe the tunnel disconnect on vscode.dev
